handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -230,42 +230,44 @@ protected void deallocate() {
         this.jdkCompatibilityMode = jdkCompatibilityMode;
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
+        final long ssl;
         try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
+            this.ssl = ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
@Scottmitch @carl-mastrangelo on second thought wouldn't it make sense to check isDestroyed first thing inside the sync block within the constructor before assigning to ssl and freeing the ssl then returning early?